gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagProcArbiterDecorator.java

```diff
@@ -0,0 +1,140 @@
+        return leaseAttemptStatus;
+      }
+    /* Otherwise leaseAttemptStatus instanceof {@link MultiActiveLeaseArbiter.LeaseObtainedStatus/NoLongerLeasingStatus}
+      and no need to do anything
+     */
+      return leaseAttemptStatus;
+    } catch (SchedulerException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected void scheduleReminderForEvent(MultiActiveLeaseArbiter.LeasedToAnotherStatus leaseStatus, long triggerEventTimeMillis)
+      throws SchedulerException {
+    // TODO: determine which ts to use
+    dagProcReminderScheduler.scheduleReminderJob(leaseStatus.getFlowAction(), leaseStatus.getEventTimeMillis());
+
+  }
+
+  @Slf4j
+  public class ReminderJob implements Job {
+    public static final String FLOW_ACTION_TYPE_KEY = "flow.actionType";
+    public static final String DAG_TASK_STREAM = "dag.taskStream";
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+      // Get properties from the trigger to create a dagAction
+      String flowName = context.getTrigger().getJobDataMap().getString(ConfigurationKeys.FLOW_NAME_KEY);
+      String flowGroup = context.getTrigger().getJobDataMap().getString(ConfigurationKeys.FLOW_GROUP_KEY);
+      String jobName = context.getTrigger().getJobDataMap().getString(ConfigurationKeys.JOB_NAME_KEY);
+      String flowId = context.getTrigger().getJobDataMap().getString(ConfigurationKeys.FLOW_EXECUTION_ID_KEY);
+      DagActionStore.FlowActionType flowActionType = DagActionStore.FlowActionType.valueOf(
+          context.getTrigger().getJobDataMap().getString(FLOW_ACTION_TYPE_KEY));
+       DagManagementTaskStreamImpl dagManagementTaskStream =
+           (DagManagementTaskStreamImpl) context.getTrigger().getJobDataMap().get(DAG_TASK_STREAM);
+
+      // TODO: add meaningful log statement
+      log.info("Reminder for job " + jobName + " in flow " + flowName + " (" + flowGroup + ") with ID " + flowId);
+
+      DagActionStore.DagAction dagAction = new DagActionStore.DagAction(flowGroup, flowName, flowId, jobName, flowActionType);
+
+       dagManagementTaskStream.addDagAction(dagAction);
+    }
+  }
+
+  public static String createDagActionReminderKey(DagActionStore.DagAction dagAction) {
+    return createDagActionReminderKey(dagAction.getFlowName(), dagAction.getFlowGroup(), dagAction.getJobName(),
+        dagAction.getFlowExecutionId(), dagAction.getFlowActionType());
+  }
+
+  public static String createDagActionReminderKey(String flowName, String flowGroup, String jobName, String flowId,
+      DagActionStore.FlowActionType flowActionType) {
+    return String.format("%s.%s.%s.%s.%s", flowName, flowGroup, jobName, flowId, flowActionType);
+  }
+
+  public static JobDetail createReminderJobDetail(DagManagementTaskStreamImpl taskStream, DagActionStore.DagAction dagAction) {
+    JobDataMap dataMap = new JobDataMap();
+    dataMap.put(ReminderJob.DAG_TASK_STREAM, taskStream);
```

Comment:
what does it mean to stuff this into a data map?  if it involves serialization, I'm not in favor, because the `DagMgmtTaskStreamImpl` could have lengthy internal queues.

given there's only one DMTSImpl per JVM, do we really need it to ride along within the reminder?

Reply:
yes, if non-`static` is an option, that's probably our solution here